lib/ansible/modules/cloud/azure/azure_rm_devtestlabartifact_facts.py

```diff
@@ -0,0 +1,244 @@
+      name: myArtifact
+'''
+
+RETURN = '''
+artifacts:
+    description: A list of dictionaries containing facts for DevTest Lab Artifact.
+    returned: always
+    type: complex
+    contains:
+        id:
+            description:
+                - The identifier of the artifact.
+            returned: always
+            type: str
+            sample: "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.DevTestLab/labs/myLab/ar
+                     tifactSources/myArtifactSource/artifacts/myArtifact"
+        resource_group:
+            description:
+                - Name of the resource group.
+            returned: always
+            type: str
+            sample: myResourceGroup
+        lab_name:
+            description:
+                - Name of the lab.
+            returned: always
+            type: str
+            sample: myLab
+        artifact_source_name:
+            description:
+                - The name of the artifact source.
+            returned: always
+            type: str
+            sample: myArtifactSource
+        name:
+            description:
+                - The name of the artifact.
+            returned: always
+            type: str
+            sample: myArtifact
+        description:
+            description:
+                - Description of the artifact.
+            returned: always
+            type: str
+            sample: Installs My Software
+        file_path:
+            description:
+                - "Artifact's path in the repo."
+            returned: always
+            type: str
+            sample: Artifacts/myArtifact
+        publisher:
+            description:
+                - Publisher name.
+            returned: always
+            type: str
+            sample: MyPublisher
+        target_os_type:
+            description:
+                - Target OS type.
+            returned: always
+            type: str
+            sample: Linux
+        title:
+            description:
+                - Title of the artifact.
+            returned: always
+            type: str
+            sample: My Software
+        parameters:
+            description:
+                - A dictionary containing parameters definition of the artifact.
+            returned: always
+            type: complex
+            sample: {}
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.devtestlabs import DevTestLabsClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMArtifactFacts(AzureRMModuleBase):
+    def __init__(self):
+        # define user inputs into argument
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            lab_name=dict(
```

Comment:
lab_name not required?

Reply:
indeed, I wonder why it wasn't caught by sanity tests.....